### PR TITLE
Promotion from AMD internal branch for 2023.Q3.2

### PR DESCRIPTION
**Pull request by chuang13**

### CODE REVIEW

File: lgc/state/PalMetadata.cpp [hunk 1]
```diff
@@ -147,8 +147,8 @@ void PalMetadata::initialize() {
   // Pre-find (or create) heavily used nodes.
   m_pipelineNode =
       m_document->getRoot().getMap(true)[Util::Abi::PalCodeObjectMetadataKey::Pipelines].getArray(true)[0].getMap(true);
-  if (!m_useRegisterFieldFormat)
-    m_registers = m_pipelineNode[".registers"].getMap(true);
+  // Add ".registers" unconditionally is a temporary workaround since PAL needs read this section for parsing.
+  m_registers = m_pipelineNode[".registers"].getMap(true);
```
Comment (nhaehnle) [inline lgc/state/PalMetadata.cpp:151]: Do we actually need this change on gpuopen?

**Review by nhaehnle — COMMENTED** [review]
Just some things I noticed while skimming the PR.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 26f6c8a90ec8108612ad35b547a81718ca3de2c0
<details><summary>CTS tests (Failed: 0/208496)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/state/PalMetadata.cpp (hunk 1 above)
Comment (ruiling) [reply lgc/state/PalMetadata.cpp:151]: This workaround has been removed internally; But the promotion release happens before that internal change. The lgc lit testing already fail for this metadata difference. I think we need to include the internal change to remove the workaround here.

Comment (xuechen417) [reply lgc/state/PalMetadata.cpp:151]: This workaround is not needed any more.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 77bfaf37b461ef5fe9d6174bb4d69859d77dfd8b
<details><summary>CTS tests (Failed: 1/208497)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 8.9, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li><details><summary>Failed: 1/69503 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.synchronization2.op.multi_queue.timeline_semaphore.write_ssbo_geometry_read_ssbo_tess_control.buffer_16384_exclusive
Stack trace: Flake
```
</p></details></li>

<li>Not Supported: 32524/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 77bfaf37b461ef5fe9d6174bb4d69859d77dfd8b
<details><summary>CTS tests (Failed: 0/208497)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — DISMISSED** [review]

**Comment by amdvlk-admin** [comment]
f9d6782053530428b5d0d2f03bfc59f391aad199 Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/shared/continuations/lib/LowerRaytracingPipeline.cpp:789:10: error: base operand of ‘->’ has non-pointer type ‘llvm::CallInst’
     CInst->replaceAllUsesWith(IsGlobal);
          ^~
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/shared/continuations/lib/LowerRaytracingPipeline.cpp:790:10: error: base operand of ‘->’ has non-pointer type ‘llvm::CallInst’
     CInst->eraseFromParent();
          ^~
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/shared/continuations/lib/LowerRaytracingPipeline.cpp: In lambda function:
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/shared/continuations/lib/LowerRaytracingPipeline.cpp:812:10: error: base operand of ‘->’ has non-pointer type ‘llvm::CallInst’
     CInst->replaceAllUsesWith(Addr);
          ^~
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/shared/continuations/lib/LowerRaytracingPipeline.cpp:813:10: error: base operand of ‘->’ has non-pointer type ‘llvm::CallInst’
     CInst->eraseFromParent();
          ^~
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/shared/continuations/lib/LowerRaytracingPipeline.cpp: In member function ‘void llvm::LowerRaytracingPipelinePassImpl::processFunctionEnd(llvm::IRBuilder<>&, llvm::LowerRaytracingPipelinePassImpl::FunctionData&, llvm::LowerRaytracingPipelinePassImpl::FunctionEndData&)’:
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/shared/continuations/lib/LowerRaytracingPipeline.cpp:1226:19: warning: unused variable ‘Callee’ [-Wunused-variable]
         if (auto *Callee = Call->getCalledFunction()) {

**Pull request closed without merging**